clang/include/clang/Basic/Builtins.td

```diff
@@ -896,6 +896,12 @@ def Launder : Builtin {
   let Prototype = "void*(void*)";
 }
 
+def StartObjectLifeTime : Builtin {
+  let Spellings = ["__builtin_start_object_lifetime"];
```

Comment:
IMO if the intent of this intrinsic is to handle polymorphic classes (which is not covered by `std::start_lifetime_as`), its name should diverge from "plain" `start_lifetime`.

Reply:
I don't have strong opinion on the name. Since this builtin doesn't strictly align with std::start_lifetime_as, I think it's better to have some divergence in the name.